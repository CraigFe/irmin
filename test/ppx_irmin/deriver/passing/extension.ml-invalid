(* This file has suffix [.ml-invalid] to avoid registering it as a library in
   Dune. Once https://github.com/mirage/irmin/pull/970 is merged, this can be
   handled as a regular file (with a preprocessing dependency). *)

let _ = [%irmin: unit]

let _ = [%irmin: (string * int) list]

let _ = [%irmin: (int, [ `Msg of string ]) result]

type foo

type a

type b

let _ = [%ty: ((int, a * b * (int[@nobuiltin])) result, foo Lazy.t) Map.t]

let _ = [%ty: a * [%e b]]
