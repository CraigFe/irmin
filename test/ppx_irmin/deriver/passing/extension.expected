let _ = let open Irmin.Type in unit
let _ = let open Irmin.Type in list (pair string int)
let _ =
  let open Irmin.Type in
    result int
      (((variant "" (fun msg -> function | `Msg x1 -> msg x1)) |~
          (case1 "Msg" string (fun x1 -> `Msg x1)))
         |> sealv)
type foo
type a
type b
let _ = Map.t (result int (triple a_t b_t int_t)) (Lazy.t foo_t)
let _ = pair a_t b
